src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -125,12 +128,28 @@ func NewDownsamplerAndWriter(
 	instrumentOpts instrument.Options,
 ) DownsamplerAndWriter {
 	scope := instrumentOpts.MetricsScope().SubScope("downsampler")
+
+	sourceTags := map[ts.SourceType]string{
+		ts.SourceTypePrometheus:  "prometheus",
+		ts.SourceTypeGraphite:    "graphite",
```

Comment:
What if someone adds a new source type? Should this mapping be updated as well? Maybe we could define all supported source types and here we could just enumerate them?

Reply:
This is mostly related to the naming dilemma above. Worst case - we will see some "unknown" labels and if we care, we will fix it.